walkeeper/src/callmemaybe.rs

```diff
@@ -0,0 +1,131 @@
+//!
+//!  Callmemaybe thread is responsible for periodically requesting
+//!  pageserver to initiate wal streaming.
+//!
+use crate::receive_wal::WalStreamInfo;
+use crate::SafeKeeperConf;
+use anyhow::Result;
+use crossbeam_channel::{Receiver, RecvTimeoutError};
+use log::*;
+use std::sync::Arc;
+use std::time::Duration;
+use tokio_postgres::NoTls;
+use zenith_utils::zid::{ZTenantId, ZTimelineId};
+
+async fn request_callback(
+    ps_addr: String,
+    listen_pg_addr: String,
+    pageserver_auth_token: Option<String>,
+    timelineid: ZTimelineId,
+    tenantid: ZTenantId,
+) -> Result<()> {
+    // Connect to the pageserver.
+    let ps_addr_split: Vec<&str> = ps_addr.split(':').collect();
```

Comment:
rewrote to use SocketAddr.
BTW, is there any reason we don't use it for all `*_addr` in SafeKeeperConf instead of String?

Reply:
I think nothing keeps us from doing that. I remember that in pageserver there are two config structures and one of them is used for toml serialization, so it may be necessary to add some custom handlers to parse socket addr from string. But worth checking. For safekeeper there is nothing similar because there is no persistent config, so we can just do it :)